src/inmanta/util/__init__.py

```diff
@@ -392,6 +392,25 @@ def action_function() -> None:
         self._scheduled[task_spec] = handle
         return task_spec
 
+    def schedule(
```

Comment:
What does this method offer? I do prefer the name, but I feel like it's just a layer on top of `add_action` that makes a tiny streamline of the interface for one type of schedules, at the cost of dropping support for the other. So now we need to call either `schedule()` or `add_action()` depending on the use case.

Reply:
To me it seems like it adds more confusion than anything else. If I see a method `schedule` I'd assume that to be the main interface, but it isn't because it only supports one of the two schedule types.

So I prefer it without, but if you feel it's useful I won't stand in the way.